org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/ui/form/AbstractForm.java

```diff
@@ -902,6 +916,28 @@ private MultiStatus ensureMultiStatus(IStatus s) {
     return ms;
   }
 
+  @Override
+  public String getNotificationBadgeText() {
+    return getNotificationBadgeStatus().map(Status::getMessage).orElse(null);
+  }
+
+  @Override
+  public void setNotificationBadgeText(String notificationBadgeText) {
+    getNotificationBadgeStatus().ifPresent(notificationBadgeStatus -> removeStatus(notificationBadgeStatus));
+    addStatus(new NotificationBadgeStatus(notificationBadgeText));
+  }
+
+  protected Optional<NotificationBadgeStatus> getNotificationBadgeStatus() {
+    final MultiStatus ms = getStatusInternal();
+    if(ms == null || !ms.containsStatus(NotificationBadgeStatus.class)){
+      return Optional.empty();
+    }
+    return ms.getChildren().stream()
+        .filter(status -> ObjectUtility.equals(status.getCode(), NotificationBadgeStatus.NOTIFICATION_BADGE_STATUS_CODE))
```

Comment:
Maybe use instanceof NotificationBadgeStatus to ensure the following cast won't fail?

Reply:
Done 👍🏼